### DISCUSSION TITLE
How to use profiler with a next.js app?

### DISCUSSION THREAD

**Discussion by ideashower** [vercel/next.js#16413] [Help] [answered]
I'd like to profile my API endpoints to help determine the source of some slow api responses and found this method that node provides: https://nodejs.org/en/docs/guides/simple-profiling/

How do I use that profiler with Next.js? 

From what I can tell, in order to use it, you need to add `--prof` to the node command that starts the server. `--prof` is not allowed in `NODE_OPTIONS` so it wasn't clear if there was a way to do it with some sort of config on the standard `next dev` command.

I tried making a custom server (per these instructions: https://nextjs.org/docs/advanced-features/custom-server) and then calling `node --prof server.js` which gets me a set of v8 log files as the server starts, but it doesn't seem to continue to generate log files after that initial startup (ie no logs seem to be generated when I hit the api endpoint).

**Comment by ideashower** [comment, marked as answer]
Nevermind, I found that I was about to get what I wanted with Chrome's profiler and using `NODE_OPTIONS='--inspect' next dev` https://medium.com/@paul_irish/debugging-node-js-nightlies-with-chrome-devtools-7c4a1b95ae27#.pmqejrn8q)

**Comment by MonstraG** [comment]
I'm here because I want to profile build process.

We're using `pnpm`, so after `pnpm install`, you'll get a `node_modules/.bin/next` sh script which does:

```
exec node "$basedir/../next/dist/bin/next" "$@"
```

at the end.

adding `--prof`  works, but that spawns 17 profile logs.

So, following [stack-overflow](https://stackoverflow.com/questions/30461821/node-with-prof-option-creating-multiple-log-files-instead-of-one-v8-log) we can add `--no-logfile-per-isolate`.
